### PR TITLE
Make unsorted activities model similar to sorted

### PR DESCRIPTION
**Pull request by lukasz-kaniowski**
- Instead of using only names for unsorted activities this change is representing model in similar way as sorted activities.
- Adds tests to check for title and url on the unsorted activities page

### CODE REVIEW

File: app/controllers/activity-view-model.js [hunk 1]
```diff
@@ -3,16 +3,14 @@ const i18n = require('i18n');
...
-          // eslint-disable-next-line no-underscore-dangle
-          title: i18n.__(`activity.${activity}.title`),
-          name: activity,
-        }));
+  activityModel(activity) {
+    return Object.assign(
+      {
+        // eslint-disable-next-line no-underscore-dangle
+        title: i18n.__(`activity.${activity.activity}.title`),
+      }, activity);
```
Comment (lukasz-kaniowski) [inline app/controllers/activity-view-model.js]: I've change this from operating on list to a single object to make it easier to spot how model for activity looks like

File: app/controllers/unsorted-activities-controller.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ router.get('/', (req, res, next) => {
     .findAllByAccountId(accountId)
     .then((sortedActivities) => {
       res.render('unsorted-activities', Object.assign({ accountId },
-        new ActivityView(getUnsortedActivityNames(sortedActivities))));
+        new ActivityModel(getUnsortedActivities(sortedActivities))));
```
Comment (nathancashmoredwp) [inline app/controllers/unsorted-activities-controller.js]: Not sure about ActivityModel as a name... for consistency if its ActivityModel I associate it with an extension of db.Model.  In this instance i still think View as a term should still be included somehow as that is still the intended purpose of this model ?

File: test/integration/unsorted-activities.spec.js [hunk 3]
```diff
@@ -22,16 +22,29 @@ describe('Unsorted activities page', () => {
...
-          .to.equal(activities.length - 2))
+    );
+
+    it('should display correct number of unsorted activities', () =>
+      expect(unsortedActivitiesPage.activityList().length)
+        .to.equal(activities.length - 2)
+    );
+
+    it('should display correct title for activity', () => {
+      expect(unsortedActivitiesPage.activityList()[0].title).to.equal('Get some work experience');
```
Comment (nathancashmoredwp) [inline test/integration/unsorted-activities.spec.js]: Bit worried about the assumption of ordering and making the association between activities and their titles.
Is it worth extracting the titles into an object that shows the relationship to the activity at the start or even checking the whole list to make it clear ?

**Comment by nathancashmoredwp** [comment]
We could even move all of the sorted and unsorted logic into the sorted-activity-model.  Rename it to activity-model and have two methods 
getSortedActivities
getUnsortedActivities

and put the filtering logic there ?  What do you think ?

**Comment by lukasz-kaniowski** [comment]
I was exploring this idea yesterday. I wanted to finish work before going home but your comment makes me think that I should try to do it.

**Comment by lukasz-kaniowski** [comment]
activity model now contains finders for sorted and unsorted by accountId. @nathancashmoredwp have a look. 
Will try to clean tests next.

**Comment by nathancashmoredwp** [comment]
👍  looking good

**Comment by lukasz-kaniowski** [comment]
d2ce24c duplicates activities data for test code. Having that boundary increase my trust in test coverage. 
Test code still rely on order of activities. I'm ok with this for now - let's deal with this if that becomes a problem.

**Pull request merged**